packages/cadl-ranch-specs/http/payload/multipart/main.tsp

```diff
@@ -16,11 +16,44 @@ model Address {
   city: string;
 }
 
+model FileSpecificContentType extends File {
+  filename: string;
+  contentType: "image/jpg";
+}
+
+model FileWithHttpPartSpecificContentTypeRequest {
+  profileImage: HttpPart<FileSpecificContentType>;
+}
+
+model FileRequiredMetaData extends File {
+  filename: string;
+  contentType: string;
+}
+
+model FileWithHttpPartRequiredContentTypeRequest {
+  profileImage: HttpPart<FileRequiredMetaData>;
+}
+
+model FileOptionalContentType extends File {
+  filename: string;
+}
+
+model FileWithHttpPartOptionalContentTypeRequest {
+  profileImage: HttpPart<FileOptionalContentType>;
+}
+
+model ComplexHttpPartsModelRequest {
+  id: HttpPart<string>;
+  address: HttpPart<Address>;
+  profileImage: HttpPart<FileRequiredMetaData>;
+  previousAddresses: HttpPart<Address[]>;
+  pictures: HttpPart<FileRequiredMetaData>[];
```

Comment:
nit, may use a slightly different File model e.g. `FileOptionalContentType`, for some diversity in this test.

Reply:
Optional content type is already covered by case `fileWithHttpPartOptionalContentType`